config/main.py

```diff
@@ -911,6 +911,38 @@ def update_sonic_environment():
             display_cmd=True
         )
 
+def cache_arp_entries():
+    success = True
+    cache_dir = '/host/config-reload'
+    click.echo('Caching ARP table to {}'.format(cache_dir))
+
+    if not os.path.exists(cache_dir):
+        os.mkdir(cache_dir)
+
+    arp_cache_cmd = '/usr/local/bin/fast-reboot-dump.py -t {}'.format(cache_dir)
+    cache_proc = subprocess.Popen(arp_cache_cmd, shell=True, text=True, stdout=subprocess.PIPE)
+    _, cache_err = cache_proc.communicate()
+    if cache_err:
+        click.echo("Could not cache ARP and FDB info prior to reloading")
+        success = False
+
+    if not cache_err:
+        fdb_cache_file = os.path.join(cache_dir, 'fdb.json')
+        arp_cache_file = os.path.join(cache_dir, 'arp.json')
+        fdb_filter_cmd = '/usr/local/bin/filter_fdb_entries -f {} -a {} -c /etc/sonic/configdb.json'.format(fdb_cache_file, arp_cache_file)
+        filter_proc = subprocess.Popen(fdb_filter_cmd, shell=True, text=True, stdout=subprocess.PIPE)
+        _, filter_err = filter_proc.communicate()
+        if filter_err:
+            click.echo("Could not filter FDB entries prior to reloading")
+            success = False
+    
+    # If we are able to successfully cache ARP table info, signal SWSS to restore from our cache
+    # by creating /host/config-reload/needs-restore
+    if success:
```

Comment:
is this required? in a similar way, swss can check if the (arp.json and fdb.json" file exists and restore, correct?

Reply:
SWSS checks for file existence, but also checks /proc/cmdline to make sure fast-reboot actually occurred. I wanted to have a similar check for config reload for redundancy.